lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -0,0 +1,36 @@
+require "pry"
+
+module RuboCop
+  module Cop
+    module RootCops
+      class FactoryFileName < Cop
+        FILE_NAME_ERROR = "Factory name should match file name (one factory per file)".freeze
+        FACTORY = %i[factory].freeze
+
+        def investigate(processed_source)
+          file_path = processed_source.buffer.name
+          @base_name = File.basename(file_path, ".rb")
+        end
+
+        def on_send(node)
+          method_name = node.method_name
+          return unless _factory_method?(method_name)
+
+          # these 2 lines turn [:factory_name] into "factory_name"
+          factory_symbol_name = *node.arguments[0]
+          factory_name = factory_symbol_name[0].id2name
+
+          if factory_name != @base_name
```

Comment:
Can you also also the same rule as we have in root-server where it's also okay if the factory name is prefixed with the file name and then a double underscore? e.g. `factories/foo.rb` could define a factory named `foo__whatever_you_want`? Sometimes we have closely related factories and it is helpful to put them in the same file.

Reply:
Sounds great. Yeah I saw a few factory files in root-server that did this so that makes sense to add it in here. Thanks!